src/ol_infrastructure/applications/ocw_site/__main__.py

```diff
@@ -0,0 +1,80 @@
+from pulumi import Config, export
+from pulumi_aws import iam, s3
+from pulumi_vault import aws
+
+from ol_infrastructure.lib.aws.iam_helper import lint_iam_policy
+from ol_infrastructure.lib.ol_types import AWSBase
+from ol_infrastructure.lib.pulumi_helper import parse_stack
+
+ocw_site_config = Config("ocw_site")
+stack_info = parse_stack()
+aws_config = AWSBase(
+    tags={
+        "OU": "open-courseware",
+        "environment": f"applications_{stack_info.env_suffix}",
+    }
+)
+
+# Create S3 buckets
+# There are two buckets for each environment (QA, Production):
+# One for the that environment's draft site (where authors test content
+# changes), and one for the environment's live site.
+# See http://docs.odl.mit.edu/ocw-next/s3-buckets
+
+draft_bucket_name = f"ocw-content-draft-{stack_info.env_suffix}"
+live_bucket_name = f"ocw-content-live-{stack_info.env_suffix}"
+
+draft_bucket = s3.Bucket(
+    draft_bucket_name,
+    bucket=draft_bucket_name,
+    tags=aws_config.tags,
+)
+
+live_bucket = s3.Bucket(
+    live_bucket_name,
+    bucket=live_bucket_name,
+    tags=aws_config.tags,
+)
+
+policy_description = (
+    "Access controls for the CDN to be able to read from the"
+    f"{stack_info.env_suffix} website buckets"
+)
+s3_bucket_iam_policy = iam.Policy(
+    f"ocw-site-{stack_info.env_suffix}-policy",
+    description=policy_description,
+    path=f"/ol-applications/ocw-site/{stack_info.env_suffix}/",
+    name_prefix="aws-permissions-",
+    policy=lint_iam_policy(
+        {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Effect": "Allow",
+                    "Action": [
+                        "s3:ListBucket*",
+                        "s3:GetObject*",
+                    ],
+                    "Resource": [
+                        f"arn:aws:s3:::{draft_bucket_name}",
+                        f"arn:aws:s3:::{draft_bucket_name}/*",
+                        f"arn:aws:s3:::{live_bucket_name}",
+                        f"arn:aws:s3:::{live_bucket_name}/*",
+                    ],
+                }
+            ],
+        },
+        stringify=True,
+    ),
+)
+
+role_name = f"ocw-site-{stack_info.env_suffix}"
+ocw_site_vault_backend_role = aws.SecretBackendRole(
```

Comment:
What's the motivation for registering the Vault role? Are we using this anywhere?

Reply:
I was just copying from the ocw_studio role. I just want to create an IAM role, however I do that -- or should I just stop after creating a policy?

Reply:
If you want to create an IAM user then you would use https://www.pulumi.com/docs/reference/pkg/aws/iam/user/, for a role you would use https://www.pulumi.com/docs/reference/pkg/aws/iam/role/

The Vault role registers a template in Vault for generating an IAM user and isn't relevant for the site content since we won't be populating the Fastly config from Vault.

Reply:
Could I just leave out the creation of the user and role? We already have an IAM user for Fastly and all we need is for the policy that I'm creating to be attached to that user eventually.

Reply:
That's fine for now, though I'd prefer to have user/credential management in code rather than manually created and floating around relying on tribal knowledge.

Reply:
OK. Commit 04fb3bf takes out the creation of the role. I was going to add a user and attach the policy to it, but I wasn't clear on a few things like the path, and thought maybe we want to revisit it because we have existing Fastly configurations with a particular access key and secret key and those don't have configuration management.